lib/configure.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Sets website configuration parameters for given bucket
+ * @param {Object} aws - AWS class
+ * @param {string} bucketName - Name of bucket to be configured
+ * @param {string} indexDocument - Path to index document
+ * @param {string} errorDocument - Path to error document
+ * @param {Object} redirectAllRequestsTo - Configuration information for redirecting all requests
+ * @param {Object[]} routingRules - Rules for routing site traffic
+ */
+function configureBucket(
+  aws,
+  bucketName,
+  indexDocument,
+  errorDocument,
+  redirectAllRequestsTo,
+  routingRules
+) {
+  let params = {
+    Bucket: bucketName,
+    WebsiteConfiguration: {}
+  };
+
+  if (redirectAllRequestsTo) {
+    params.WebsiteConfiguration.RedirectAllRequestsTo = {};
+
+    let hostName = redirectAllRequestsTo.hostName;
+
+    params.WebsiteConfiguration.RedirectAllRequestsTo.HostName = redirectAllRequestsTo.hostName;
+
+    if (redirectAllRequestsTo.protocol) {
+      params.WebsiteConfiguration.RedirectAllRequestsTo.Protocol = redirectAllRequestsTo.protocol;
+    }
+  } else {
+    params.WebsiteConfiguration.IndexDocument = { Suffix: indexDocument };
+    params.WebsiteConfiguration.ErrorDocument = { Key: errorDocument };
+  }
+
+  if (routingRules) {
+    params.WebsiteConfiguration.RoutingRules = [];
+
+    routingRules.forEach(r => {
+      const routingRule = {
+        Redirect: {}
+      };
+      const redirectProps = [
+        'hostName',
+        'httpRedirectCode',
+        'protocol',
+        'replaceKeyPrefixWith',
+        'replaceKeyWith'
+      ];
+      redirectProps.forEach(p => {
+        if (r.redirect[p]) {
+          if (p === 'httpRedirectCode') {
+            r.redirect[p] = r.redirect[p].toString();
+          }
+          routingRule.Redirect[p.charAt(0).toUpperCase() + p.slice(1)] = r.redirect[p];
```

Comment:
Yup, the issue here is that AWS expects PascalCase but the properties coming in from our serverless.yml are camelCase. I suppose that we could instead add code before this to convert everything to PascalCase, but I;m not sure that's better. I'll def add a comment though

Reply:
Devil's advocate here - Can we just have the properties be PascalCase to avoid the complexity here?

I guess the trade off is that it would be a less-consistent user experience if we made them use PascalCase some places and camelCase elsewhere. I think this is a good solution - just add the comment to explain and it should be good.

Reply:
Yeah, we could switch our API to use PascalCase, but I think consistency with ourselves is worth a weird extra line of code. I'll add the comment